kubespawner/spawner.py

```diff
@@ -2185,7 +2185,9 @@ async def poll(self):
         just Falsy, to determine that the pod is still running.
         """
 
-        await self._start_watching_pods()
+        reflector = await self._start_watching_pods()
+        if not reflector.first_load_future.done():
+            await reflector.first_load_future
```

Comment:
@minrk I think this makes sense, but would appreciate your review help.

I'm especially thinking about if we should put this code in `_start_watching_pods` or `_start_reflector`.

Currently, `self._start_watching_pods` is awaited from `_start`, `stop`, and `poll`, where `_start` also awaits `await self._start_watching_events`.

Should do the await on `first_load_future` if needed from `_start_reflector`?

Reply:
I think awaiting it in `_start_reflector` makes sense. If I were to guess based on my foggy memory of the distant past, I think perhaps `_start_reflector` maybe used to have a requirement to not be async, so it couldn't wait for things? That doesn't appear to be the case anymore.